### PR TITLE
gohar94's C++ Strings Goodie

### PR DESCRIPTION
**Pull request by gohar94**
# Goodie Pull Request Template

**What does your Instant Answer do?**
Shows basic C++ String operations

**What problem does your Instant Answer solve (Why is it better than organic links)?**
Quickly shows relevant chunks of code, easy to copy-paste

**What is the data source for your Instant Answer? (Provide a link if possible)**
http://www.cplusplus.com/reference/string/string/

**Why did you choose this data source?**
Authentic C++ docs

**Are there any other alternative (better) data sources?**
This is the most authentic

**What are some example queries that trigger this Instant Answer?**
c++ strings

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
Programmers, people new to coding and learning

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
https://duck.co/ia/view/cpp_strings_cheat_sheet

**Which existing Instant Answers will this one supercede/overlap with?**
None

**Are you having any problems? Do you need our help with anything?**
No

**Where did you hear about DuckDuckHack? (For first time contributors)**
StackOverflow

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
screen shot 2015-08-28 at 4 41 50 am

## Checklist
Please place an 'X' where appropriate.

```
- [] Added metadata and attribution information (https://duck.co/duckduckhack/metadata)
- [] Wrote test file and added to t/ directory (https://duck.co/duckduckhack/goodie_tests)
- [] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
- [] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
- [] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 10

    - Windows 7
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 8
        - [] IE 9
        - [] IE 10

    - Windows XP
        - [] IE 7
        - [] IE 8

    - Mac OSX
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] Safari

    - iOS (iPhone)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - iOS (iPad)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - Android
        - [] Firefox
        - [] Native Browser
        - [] Google Chrome
        - [] Opera

```

-----
IA Page: https://duck.co/ia/view/cpp_strings_cheat_sheet

### CODE REVIEW

File: lib/DDG/Goodie/IsAwesome/gohar94.pm [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package DDG::Goodie::IsAwesome::gohar94;
```
Comment (moollaza) [inline lib/DDG/Goodie/IsAwesome/gohar94.pm]: Please remove this from the PR. PR's should only contain one new Instant Answer. As well, we're no longer accepting IsAwesome Goodie Pull Requests, they're really meant for you to build as a learning experience.

File: share/goodie/cheat_sheets/json/c++-strings.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{
+    "id": "c++_strings_cheat_sheet",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/c++-strings.json]: please change to `cpp_strings_cheat_sheet`

**Comment by gohar94** [comment]
Made the changes. Please have a look.

File: share/goodie/cheat_sheets/json/c++-strings.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{
+    "id": "cpp_strings_cheat_sheet",
+    "name": "C++ Strings",
+    "description": "List of basic String operations in C++",
+    "metadata": {
+        "sourceName": "C++ Reference",
+        "sourceUrl": "http://www.cplusplus.com/reference/string/string/"
+    },
+    "keylate_type": "code",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/c++-strings.json]: typo: should be `template_type`

**Comment by gohar94** [comment]
Extremely sorry for the typo. Has been fixed.

**Comment by moollaza** [comment]
@gohar94 This LGTM :+1: Apologies for the delay! 

Thanks a lot for contributing (we really appreciate it :smiley:). We invite you to stick around and comment/help on other PR's & Issues or even submit an Instant Answer!

**Comment by moollaza** [comment]
@gohar94 congrats! It's live: https://duckduckgo.com/?q=c%2B%2B+strings+cheat+sheet&ia=cheatsheet

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

**Pull request merged**